compiler/rustc_passes/src/naked_functions.rs

```diff
@@ -274,12 +274,25 @@ impl<'tcx> CheckInlineAssembly<'tcx> {
         }
 
         if !asm.options.contains(InlineAsmOptions::NORETURN) {
+            let last_span = asm
+                .operands
+                .last()
+                .map_or_else(|| asm.template_strs.last().unwrap().2, |op| op.1)
+                .shrink_to_hi();
+
             struct_span_err!(
                 self.tcx.sess,
                 span,
                 E0787,
                 "asm in naked functions must use `noreturn` option"
             )
+            .span_suggestion(
+                last_span,
+                "consider specifying that the asm block is responsible \
+                for returning, if desired",
```

Comment:
I would suggest qualifying what kind of return we have in mind by saying that "asm block is responsible for returning *from function*", to avoid ambiguity with returning as used in `noreturn` option, which is essentially the opposite of that. `:-(`.

The "if desired" feels redundant, especially with "consider" already present.

Reply:
"if desired" was meant to indicate that it could also be truly non-returning (exit the process or whatnot) but I suppose that part is ambiguous too, will be sure to fix that

Reply:
Actually I decided to just drop that, I think the option for actions which truly never return is enough of a given that it's worth omitting for clarity/brevity